### PR TITLE
Add release notes for 2.2.0

### PR DESCRIPTION
**Pull request by bunchesofdonald**
### Summary
Add release notes for 2.2.0

### Checklist

This pull request is:

- [x] A documentation / typographical error fix
	- No tests or issue needed
- [ ] A short code fix
	- Please reference the related issue by including "closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a bug report issue 
	- Please include tests. One-line fixes without tests will not be accepted unless it's related to the documentation only.
- [ ] A new feature implementation
	- Please reference the related issue by including "closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a feature enhancement issue 
	- Please include tests
    - Please make sure that your QA steps are both thorough and easy to reproduce by somebody with limited knowledge of the feature that you are submitting

**Happy engineering!**

### CODE REVIEW

**Review by anna-geller — CHANGES_REQUESTED** [review]
Looks good but I would remove the new contributors section - it's Evan 😄

**Review by anna-geller — APPROVED** [review]

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,5 +1,30 @@
...
+* Disallowed extra attribute initialization on Deployments in #6505
+* Updated `deployment build` to raise an informative error if two infrastructure configs are provided in #6504
+* Fixed calling async subflows from sync parents in #6514
+
+### Documentation
+* Removed references to 2.0.5 in #6473
+* Updated deployment concepts to reflect changes in 2.1.1 in #6490
+* Fixed typos in blocks.md in #6508
+* Fixed typos in README in #6498
+* Moved registering blocks to its own heading in #6512
```
Comment (madkinsz) [inline RELEASE-NOTES.md:23]: I'd omit these generally

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,5 +1,30 @@
...
+### Exciting New Features 🎉
+* Added automatic detection of static arguments to `Task.map` in #6513
+* Using new orion-design flow-run-filter composition in #6483
+
+### Fixes
+* Updated deployment flow run retry settings with runtime values in #6489
+* Updated media queries for flow-run-filter in #6484
+* Added `empirical_policy` to flow run update route in #6486
+* Updated flow run policy retry settings to be nullable in #6488
+* Disallowed extra attribute initialization on Deployments in #6505
```
Comment (madkinsz) [inline RELEASE-NOTES.md:14]:
```suggestion
* Disallowed extra attribute initialization on `Deployment` objects in #6505
```

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,5 +1,30 @@
 # Prefect Release Notes
 
+## 2.2.0
+
+### Exciting New Features 🎉
+* Added automatic detection of static arguments to `Task.map` in #6513
+* Using new orion-design flow-run-filter composition in #6483
```
Comment (madkinsz) [inline RELEASE-NOTES.md:7]: What does this mean to a user? Verb doesn't match other format.

Comment (bunchesofdonald) [reply RELEASE-NOTES.md]: Looking closer at that PR it looks like it was actually just some code reorganization, so I just removed the bullet.

**Pull request merged**